### PR TITLE
proof of concept

### PR DESCRIPTION
**Pull request by AndyAyersMS**
FYI @EgorBo @jakobbotsch 

This gets the `?:` case Egor was discussing and maybe the local address one Jakob was looking at.

Not sure this is the best approach, but it does some interesting things. It leverage the local threading so hopefully isn't' too costly.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
Sample diff

image

**Comment by jakobbotsch** [comment]
> maybe the local address one Jakob was looking at

To handle these cases we need to do something before or during local morph, otherwise we will still end up with address exposure.

**Comment by AndyAyersMS** [comment]
This is like an anti-head-merge... if we were to take this along with head merge we'd need to figure out how not to undo what we just did.

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -7039,3 +7039,337 @@ PhaseStatus Compiler::fgTailMerge()
...
+    {
+        madeChanges |= fgTailDuplicateBlock(block);
+    }
+
+    return madeChanges ? PhaseStatus::MODIFIED_EVERYTHING : PhaseStatus::MODIFIED_NOTHING;
+}
+
+bool Compiler::fgTailDuplicateBlock(BasicBlock* const block)
+{
+    // We only consider blocks with muliple preds without
```
Comment (danmoseley) [inline src/coreclr/jit/fgopt.cpp:7091]:
```suggestion
    // We only consider blocks with multiple preds without
```

**Comment by jakobbotsch** [comment]
> This is like an anti-head-merge

Isn't this more like an anti-tail-merge? It duplicates the same statement at the end of multiple predecessors, so those statements candidates for tail merging.

**Comment by AndyAyersMS** [comment]
> > This is like an anti-head-merge
> 
> Isn't this more like an anti-tail-merge? It duplicates the same statement at the end of multiple predecessors, so those statements are candidates for tail merging.

Yes, you're right. Actually an anti-head-merge (head duplication? I guess) might be interesting to consider, if we think the successor blocks have more interesting context than the current block.

**Pull request closed without merging**